src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -709,3 +737,66 @@ def self_test(self):
 
         c = orchestrator.TrivialReadCompletion(result=True)
         assert c.has_result
+
+    @_write_cli('upgrade check',
+                'name=image,type=CephString,req=false',
+                desc='Check service versions vs available and target containers')
+    def _upgrade_check(self, image=None):
+        # get service state
+        completion = self.describe_service()
+        self._orchestrator_wait([completion])
+        orchestrator.raise_if_exception(completion)
+        services = completion.result
+
+        r = {
+            'global_target_image': image,
```

Comment:
With this PR, we're making it mandatory to use containers for using the orchestrator. This is of course true for all activity developed orchestrators, but adding this here makes it hard to create new orchestrators. Would it hurt to move this logic to mgr/ssh and just expose upgrage_check in the api?